### PR TITLE
Tool to extract documentation into md.

### PR DESCRIPTION
**Pull request by meowjesty**
- Issue: [#69](https://github.com/metalbear-co/mirrord.dev/issues/69)

1. Introduce tool (`medschool`) to auto-generate documentation;
2. Update mirrord/config docs to match what's in mirrord.dev;
3. Update mirrord-schema with these new docs;

P.S. as the docs are just a transfer from what we have in mirrord.dev, it might be easier to review by looking at a diff of the auto-generated docs and the previously available docs there.

### CODE REVIEW

**Review by eyalb181 — APPROVED** [review]

File: medschool/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,550 @@
...
+                        }
+                    }
+                    _ => None,
+                })
+        })
+        .collect::<BTreeSet<_>>();
+
+    // The types that have been rebuild after following each field "reference".
+    let mut new_types = Vec::with_capacity(8);
+
```
Comment (infiniteregrets) [inline medschool/src/main.rs:396]: why is the capacity hardcoded to 8

File: medschool/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,550 @@
...
+                            })
+                        } else {
+                            None
+                        }
+                    }
+                    _ => None,
+                })
+        })
+        .collect::<BTreeSet<_>>();
+
```
Comment (infiniteregrets) [inline medschool/src/main.rs:393]: can you please document the reason why you use btreeset/btreemap over hashmap

File: medschool/src/main.rs (hunk 1 above)
Comment (meowjesty) [reply medschool/src/main.rs:396]: Oh snap, some of these were just random values.

File: medschool/src/main.rs (hunk 2 above)
Comment (meowjesty) [reply medschool/src/main.rs:393]: Yup, can you spot more places where docs are lacking?

Will open a PR with whatever needs tidying up.

Comment (infiniteregrets) [reply medschool/src/main.rs:393]: sure will let you know

Comment (meowjesty) [reply medschool/src/main.rs:393]: Added a brief doc for this in the newest PR.

**Pull request merged**